### PR TITLE
RPC - add method to call API, add method for stopping staking

### PR DESCRIPTION
**Pull request by mikedennis**
Added a method to funnel api method calls through RPC. When running in docker it can be challenging to expose localhost calls externally so this lets you call api methods via RPC so you only have expose the RPC endpoint.

I had some challenges around the RPC middleware and passing JSON as parameters to RPC so to avoid that I required the JSON to be encoded in Base64 so the RPC doesn't try to deserialize it.

We already had a startstaking RPC method just added the corresponding stopstaking.

### CODE REVIEW

**Comment by mikedennis** [comment]
I get this error when I try to assign anything except localhost. System.Net.Sockets.SocketException (99): Cannot assign requested address

**Comment by zeptin** [comment]
That's strange - it can't be a limitation of docker, because the internal testnet uses docker & the API extensively. I also use `-apiuri` on my own public nodes with no issues, although those aren't in docker. What does `ifconfig` return when run from the docker container's shell?

**Comment by mikedennis** [comment]
Ok I got it to work. Thanks!! I was putting in the docker0 ip which I guess must just be a gateway. Once I got the actual container ip it worked like a charm. Awesome. Still wonder whether this RPC method might have value though. Api has no auth so rely totally on ip whitelisting but maybe the proper fix would be to optionally allow some auth on api?

File: src/Stratis.Bitcoin.Features.Miner/Controllers/StakingRPCController.cs [hunk 1]
```diff
@@ -48,12 +48,24 @@ public StakingRpcController(IFullNode fullNode, ILoggerFactory loggerFactory, IW
...
+            return true;
+        }
+
         /// <summary>
         /// Starts staking a wallet.
         /// </summary>
         /// <param name="walletName">The name of the wallet.</param>
         /// <param name="walletPassword">The password of the wallet.</param>
-        /// <returns></returns>
+        /// <returns>true.</returns>
```
Comment (quantumagi) [inline src/Stratis.Bitcoin.Features.Miner/Controllers/StakingRPCController.cs]: <c>true</c>. Can this method be made `void`. ?

**Comment by zeptin** [comment]
Just wondering what the general opinion is on the `callapi` RPC and whether it is still needed.

**Comment by bokobza** [comment]
I'm not too keen on the funnelling as it feels a bit hacky.
If there's something you need from the API but you can't expose the api, then you can add an RPC method that does the same thing, or change where your node is deployed.
That's only my view obviously.

**Comment by bokobza** [comment]
So @mikedennis can you please remove the callapi method?
Also look at the cli where we can call the api via the command line.

**Pull request merged**